packages/fiori-elements-writer/src/data/defaults.ts

```diff
@@ -11,8 +11,10 @@ import { getBaseComponent, getUi5Libs } from './templateAttributes';
 export function setDefaultTemplateSettings<T>(template: Template<T>): T {
     const templateSettings = template.settings;
     if (template.type === TemplateType.AnalyticalListPage) {
-        const alpSettings = template.settings as Partial<ALPSettingsV2>;
+        const alpSettings = template.settings as Partial<ALPSettingsV2 & ALPSettingsV4>;
 
+        // Assign defaults for properties of all ALP settings interfaces.
+        // Since only referenced properties are written by templates this has no side-effects.
```

Comment:
I am not in favor of this statement and the following implementation. Yes, right now, our templating explicitly takes the properties required, however, in other templates e.g. in `fe-fpm-writer` we also use a `JSON.stringify` in one place to set properties in the `manifest.json`. Maybe someone else will use the very same in the future to serialize the settings in the ALP templates resulting in invalid generated manifests. Of course, I would expect our test cases to catch it but it wouldn't change the fact that we add wrong settings here that just by knowledge of other parts of code don't create a problem.
Please look at the function as an encapsulated unit that should behave as it would be expected.

Reply:
OK, the alternative is rather ugly casts and type guards (which is how I originally implemented), but Ill update for your comment.

Reply:
Maybe I am missing something but I'd assumed it is 'just' another nested `if/else` with an `Object.assign` in each block?

Reply:
Almost yes - https://github.com/SAP/open-ux-tools/pull/366/commits/8f50dcb50162348bd2ea5b30b519f6fde76eba8a#diff-ff4edbeac8e42e9c7ce005e9ba864236b8e5909f154fde419beb25fa1098fa06R23-R50

Reply:
Updated to use odata version as discussed.

Reply:
Thank you!